pkg/scheduler/framework/session.go

```diff
@@ -387,8 +389,23 @@ func (ssn *Session) GetUnschedulableAndUnresolvableNodesForTask(task *api.TaskIn
 // - Unschedulable
 // - UnschedulableAndUnresolvable
 // - ErrorSkipOrWait
-func (ssn *Session) PredicateForAllocateAction(task *api.TaskInfo, node *api.NodeInfo) error {
-	err := ssn.PredicateFn(task, node)
+func (ssn *Session) PredicateForAllocateAction(task *api.TaskInfo, node *api.NodeInfo, cacheEnable bool) error {
+	var statusSets api.StatusSets
+	if node.Allocatable.MaxTaskNum <= len(ssn.NodeMap[node.Name].Pods) {
+		statusSets = append(statusSets, &api.Status{Code: api.Unschedulable, Reason: api.NodePodNumberExceeded})
+		return api.NewFitErrWithStatus(task, node, statusSets...)
+	}
+	var (
+		err error
+		ok  bool
+	)
+	if cacheEnable {
+		err, ok = util.GetPredicateCache(task.Job, task.UID, node.Name, node.HashValue)
```

Comment:
we can not simply use a cache here. For example, pod affinity/antiaffinity need to be re-considered.

Reply:
I have added task hash which will consider the change of pod affinity/antiaffinity